server/src/instant/scripts/daily_metrics.clj

```diff
@@ -0,0 +1,127 @@
+                     ": Active Devs: **" distinct_users
+                     "**, Active Apps: **" distinct_apps
+                     "**")]
+    (discord/send! config/discord-metrics-channel-id message)))
+
+(defn insert-new-activity
+  "Insert new transactions into the daily_app_transactions table.
+  This is intended to run daily to speed up monthly metrics generation."
+  ([] (insert-new-activity (aurora/conn-pool)))
+  ([conn]
+   (sql/do-execute! conn
+                    ["WITH date_range AS (
+                    SELECT
+                      COALESCE(MAX(date) + INTERVAL '1 day', '2022-01-01') AS last_seen_date,
+                      CURRENT_DATE - INTERVAL '1 day' AS max_date
+                    FROM daily_app_transactions
+                  ),
+                  earliest_transaction_per_app AS (
+                    SELECT
+                      app_id,
+                      MIN(created_at) AS earliest_date,
+                      MIN(created_at) + INTERVAL '7 days' AS earliest_date_plus_7
+                    FROM
+                      transactions
+                    GROUP BY
+                      app_id
+                  ),
+                  new_transactions AS (
+                    SELECT
+                       DATE(t.created_at) as date,
+                       t.app_id,
+                       eta.earliest_date_plus_7 as active_date,
+                       CASE  WHEN t.created_at > eta.earliest_date_plus_7 THEN true ELSE false END as is_active,
+                       count(*)
+                    FROM transactions t
+                    JOIN earliest_transaction_per_app eta ON t.app_id = eta.app_id
+                    CROSS JOIN date_range dr
+                    WHERE t.created_at > dr.last_seen_date and t.created_at < dr.max_date AND DATE(t.created_at) NOT IN ('2024-10-28')
+                    GROUP BY 1, 2, 3, 4
+                  )
+                  INSERT INTO daily_app_transactions (date, app_id, is_active, count)
+                  SELECT date, app_id, is_active, count
+                  FROM new_transactions;"])))
+
+(defn daily-job!
+  ([] (daily-job! (date/numeric-date-str (-> (LocalDate/now) (.minusDays 1)))))
+  ([date-str]
+   (grab/run-once!
+    (str "daily-metrics-" date-str)
+    (fn []
+      (insert-new-activity)
+      (let [stats (get-daily-actives date-str)]
+        (send-discord! stats date-str))))))
+
+(comment
+  (daily-job! "2024-10-04"))
+
+(defn period []
+  (let [now (date/pst-now)
+        nine-am-pst (-> now
+                        (.withHour 9)
+                        (.withMinute 0))
+        periodic-seq (chime-core/periodic-seq
+                      nine-am-pst
+                      (Period/ofDays 1))]
+
+    (->> periodic-seq
+         (filter (fn [x] (.isAfter x now))))))
+
+(defn start []
+  (log/info "Starting daily metrics daemon")
+  (def schedule (chime-core/chime-at (period) daily-job!)))
+
+(defn stop []
+  (when schedule
```

Comment:
Since schedule is a `def`, it will always be truthy here.

```suggestion
  (when (bound? #'schedule)
```

Ran into the same problem with https://github.com/instantdb/instant/blob/main/server/src/instant/reactive/invalidator.clj#L425